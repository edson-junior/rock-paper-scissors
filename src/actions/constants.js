export const initialState = {
  user: 0,
  computer: 0,
  matches: 0,
  result: null,
};
export const ROCK = 'ROCK';
export const PAPER = 'PAPER';
export const SCISSORS = 'SCISSORS';
