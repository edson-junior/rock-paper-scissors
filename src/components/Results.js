import React from 'react';
import PropTypes from 'prop-types';

/**
 * Render the redux state in the component.
 * @param {object} state - The redux initial state.
 * @returns {*} The rendered component.
 */
const Results = ({ store }) => (
  <div>
    <h3>Result: You win!</h3>

    <div>
      <h3>Score</h3>
      <p>User: { store.user }</p>
      <p>Computer: { store.computer }</p>
      <p>Matches: { store.matches }</p>
    </div>
  </div>
);

Results.propTypes = {
  store: PropTypes.shape({}).isRequired,
};

export default Results;
