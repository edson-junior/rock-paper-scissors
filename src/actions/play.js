import * as constants from './constants';

export const rock = rock => {
  type: constants.ROCK,
  rock
}

export const paper = paper => {
  type: constants.PAPER,
  paper
}

export const scissors = scissors => {
  type: constants.SCISSORS,
  scissors
}

