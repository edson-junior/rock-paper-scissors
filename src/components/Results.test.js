import React from 'react';
import Results from './Results';
import * as constants from '../actions/constants';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';

configure({ adapter: new Adapter() });

const initialState = constants.initialState;
const middlewares = [];
const mockStore = configureStore(middlewares);
const store = mockStore(initialState);
const props = { store };

const results = shallow(<Results {...props} />);

describe('Results', () => {
  it('should render properly', () => {
    expect(toJson(results)).toMatchSnapshot();
  });
});
