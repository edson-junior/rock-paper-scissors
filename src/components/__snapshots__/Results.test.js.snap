// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should render properly 1`] = `
<div>
  <h3>
    Result
  </h3>
  <p />
  <div>
    <h3>
      Score
    </h3>
    <p>
      User: 
    </p>
    <p>
      Computer: 
    </p>
    <p>
      Matches: 
    </p>
  </div>
</div>
`;
